### DISCUSSION TITLE
Questions about opentable

### DISCUSSION THREAD

**Discussion by kaspesi** [21Bruce/resolved-bot#36] [General]
See this works for resy which is super cool.  Was wondering if opentable api had limitations that prevented its implementation, or it just wasn't in the scope of this project.  Really cool project would love to discuss / see if you could use any help.

**Comment by 21Bruce** [comment]
Atm the bot doesn't work or at least has a lot of unresolved bug reports. As for your comment about opentable, we actually do have an opentable API in the source tree but it can't be integrated with the rest of the bot. This is due to their use of 2fac. One improvement would be to change the current bot algorithms to do token refreshes instead of multiple logins. In this scheme we'd have the user log in only when no refresh token is available and then maintain a timer for when our auth token expires to refresh it, like how you stay logged in to these services with a browser. If you want, you can implement that, and I'll review and commit the changes as they come in, but atm the other dev and I have more or less moved on.
